src/decoder_fmmidi.cpp

```diff
@@ -66,21 +69,33 @@ bool FmMidiDecoder::Open(FILE* file) {
 	}
 	seq->rewind();
 
-	ticks_per_sec = (float)seq->get_division() / tempo * 1000000;
+	tempo.emplace_back(this, midi_default_tempo);
 
 	return true;
 }
 
 bool FmMidiDecoder::Seek(size_t offset, Origin origin) {
+	assert(!tempo.empty());
+
 	if (offset == 0 && origin == Origin::Begin) {
 		mtime = seq->rewind_to_loop();
-		mtime_last_tempo_change = mtime_ltc_loopstart;
-		ticks_last_tempo_change = ticks_ltc_loopstart;
+		// Throw away all tempo data after the loop point
+		auto it = tempo.end();
+		while (it > tempo.begin()) {
+			--it;
+			if (it->mtime <= mtime) {
+				++it;
+				break;
+			}
+		}
+		if (it != tempo.end()) {
+			tempo.erase(it, tempo.end());
+		}
```

Comment:
@fmatthew5876 
Is there a more idiomatic way to do this?
I search from the end as I know that all elements to delete are at the end but "erase" with reverse iterators gives a compiler error :(